### PR TITLE
Add ability to render preview browsers (introduced in BCD 4.0.0)

### PR DESCRIPTION
**Pull request by Elchi3**
(This will fail as it depends on BCD 4.0.0 which will include https://github.com/mdn/browser-compat-data/pull/10334.
Edit: Marked as draft to prevent it from being merged early. This is ready for review but needs a coordinated merge with the release of BCD 4.0.0.)

For a long time BCD contributors and browser vendors have asked for a way to include preview browsers, that is, data for Firefox Nightly, Chrome Canary, and Safari Technical Preview. In https://github.com/mdn/browser-compat-data/pull/10334 we modeled and agreed on a way forward: The BCD data model will be extended to include a "preview" enum to indicate support in these browsers.

So, BCD 4.0.0 can include compat data like this:

<details>
<summary>Expand</summary>

```json
{
  "api": {
    "ANGLE_instanced_arrays": {
      "__compat": {
        "mdn_url": "https://developer.mozilla.org/docs/Web/API/ANGLE_instanced_arrays",
        "support": {
          "chrome": {
            "version_added": "preview"
          },
          "chrome_android": {
            "version_added": "32"
          },
          "edge": {
            "version_added": "12"
          },
          "firefox": {
            "version_added": "preview"
          },
          "firefox_android": {
            "version_added": true
          },
          "ie": {
            "version_added": "11"
          },
          "opera": {
            "version_added": "19"
          },
          "opera_android": {
            "version_added": "19"
          },
          "safari": {
            "version_added": "preview"
          },
          "safari_ios": {
            "version_added": "8"
          },
          "samsunginternet_android": {
            "version_added": "2.0"
          },
          "webview_android": {
            "version_added": "4.4"
          }
        },
        "status": {
          "experimental": false,
          "standard_track": true,
          "deprecated": false
        }
      },
      "drawArraysInstancedANGLE": {
        "__compat": {
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ANGLE_instanced_arrays/drawArraysInstancedANGLE",
          "support": {
            "chrome": {
              "version_added": "32"
            },
            "chrome_android": {
              "version_added": "32"
            },
            "edge": {
              "version_added": "12"
            },
            "firefox": {
              "version_added": "47"
            },
            "firefox_android": {
              "version_added": true
            },
            "ie": {
              "version_added": "11"
            },
            "opera": {
              "version_added": "19"
            },
            "opera_android": {
              "version_added": "19"
            },
            "safari": {
              "version_added": "8",
              "version_removed": "preview"
            },
            "safari_ios": {
              "version_added": "8"
            },
            "samsunginternet_android": {
              "version_added": "2.0"
            },
            "webview_android": {
              "version_added": "4.4"
            }
          },
          "status": {
            "experimental": false,
            "standard_track": true,
            "deprecated": false
          }
        }
      },
      "drawElementsInstancedANGLE": {
        "__compat": {
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ANGLE_instanced_arrays/drawElementsInstancedANGLE",
          "support": {
            "chrome": [
              {
                "version_added": "preview"
              },
              {
                "alternative_name": "foo",
                "version_added": "39",
                "version_removed": "45"
              }
            ],
            "chrome_android": {
              "version_added": "32"
            },
            "edge": {
              "version_added": "12"
            },
            "firefox": {
              "version_added": "47"
            },
            "firefox_android": {
              "version_added": true
            },
            "ie": {
              "version_added": "11"
            },
            "opera": {
              "version_added": "19"
            },
            "opera_android": {
              "version_added": "19"
            },
            "safari": {
              "version_added": "8"
            },
            "safari_ios": {
              "version_added": "8"
            },
            "samsunginternet_android": {
              "version_added": "2.0"
            },
            "webview_android": {
              "version_added": "4.4"
            }
          },
          "status": {
            "experimental": false,
            "standard_track": true,
            "deprecated": false
          }
        }
      },
      "vertexAttribDivisorANGLE": {
        "__compat": {
          "mdn_url": "https://developer.mozilla.org/docs/Web/API/ANGLE_instanced_arrays/vertexAttribDivisorANGLE",
          "support": {
            "chrome": [
              {
                "prefix": "webkit",
                "version_added": "15"
              },
              {
                "version_added": "preview"
              }
             ],
            "chrome_android": {
              "version_added": "32"
            },
            "edge": {
              "version_added": "12"
            },
            "firefox": {
              "version_added": "47"
            },
            "firefox_android": {
              "version_added": true
            },
            "ie": {
              "version_added": "11"
            },
            "opera": {
              "version_added": "19"
            },
            "opera_android": {
              "version_added": "19"
            },
            "safari": {
              "version_added": "8"
            },
            "safari_ios": {
              "version_added": "8"
            },
            "samsunginternet_android": {
              "version_added": "2.0"
            },
            "webview_android": {
              "version_added": "4.4"
            }
          },
          "status": {
            "experimental": false,
            "standard_track": true,
            "deprecated": false
          }
        }
      }
    }
  }
}
```
</details>

This PR aims to extend the BCD renderer so that the data can be displayed like this:

Bildschirmfoto 2021-06-09 um 14 17 32

### CODE REVIEW

File: client/src/document/ingredients/browser-compatibility-table/atoms/_bc-supports.scss [hunk 1]
```diff
@@ -84,3 +84,7 @@ $bc-partial-support-cross: rgba(144, 99, 18, 0.2);
 .bc-supports-unknown {
   background-color: $neutral-550;
 }
+
+.bc-supports-preview {
+  background-color: $primary-500;
```
Comment (escattone) [inline client/src/document/ingredients/browser-compatibility-table/atoms/_bc-supports.scss:89]: This color doesn't exist any longer in the latest `@mdn/minimalist`. I think the closest equivalent in the latest color palette is `$mdn-blue100`? (cc @schalkneethling)
```suggestion
  background-color: $mdn-blue100;
```

**Review by escattone — CHANGES_REQUESTED** [review]
Here's what I did to test this:
1. Fetched the `Elchi3:preview-browsers` branch of ` mdn/browser-compat-data` locally
2. Fetched the `Elchi3:preview-browsers` branch of ` mdn/yari` locally, and re-based it on the latest `main` branch of `mdn/yari` to prevent an error related to a popularities file.
3. Changed [line 52](https://github.com/mdn/yari/blob/67b3119221d4cc29f511da2ba963d41df9957c9e/package.json#L52) in `package.json` to use my local `Elchi3:preview-browsers` branch of ` mdn/browser-compat-data` so I'm using the new BCD 4.0.0 data and format:
    ```js
    "@mdn/browser-compat-data": "file:../browser-compat-data",
    ```
4. Copied the `api/ANGLE_instanced_arrays.json` from the introduction into my local `mdn/browser-compat-data`
5. Ran the tests (looked good)
6. Ran a complete build of the English docs (no errors)
7. Checked the BCD table that was built locally (looked good!)

image

I did have to make one minor change that's also necessary to change in this PR (see below), but otherwise, this looks good to me!

**Comment by Elchi3** [comment]
Thank you so much Ryan! 🎉

**Review by escattone — APPROVED** [review]
Thanks @Elchi3! This is ready to go once version `4.0.0` of `@mdn/browser-compat-data` is released and Yari is updated to use that version. 🎉 🚀 

FYI, the tests should all pass once Yari is updated to use version `4.0.0` of `@mdn/browser-compat-data`.

**Comment by ddbeck** [comment]
📣 [BCD v4.0.0 just dropped](https://github.com/mdn/browser-compat-data/releases/tag/v4.0.0)! 📣

When Yari upgrades to v4.0.0, this PR needs to go along with it.

**Comment by peterbe** [comment]
What are some good pages to test it on?

**Pull request merged**